android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.kt

```diff
@@ -0,0 +1,1017 @@
+ */
+@ReactModule(name = CameraRollModule.NAME)
+class CameraRollModule(reactContext: ReactApplicationContext?) :
+    ReactContextBaseJavaModule(reactContext) {
+    override fun getName(): String {
+        return NAME
+    }
+
+    /**
+     * Save an image to the gallery (i.e. [MediaStore.Images]). This copies the original file
+     * from wherever it may be to the external storage pictures directory, so that it can be scanned
+     * by the MediaScanner.
+     *
+     * @param uri     the file:// URI of the image to save
+     * @param promise to be resolved or rejected
+     */
+    @ReactMethod
+    fun saveToCameraRoll(uri: String?, options: ReadableMap, promise: Promise) {
+        SaveToCameraRoll(reactApplicationContext, Uri.parse(uri), options, promise)
+            .executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
+    }
+
+    private class SaveToCameraRoll(
+        context: ReactContext,
+        uri: Uri,
+        options: ReadableMap,
+        promise: Promise
+    ) : GuardedAsyncTask<Void?, Void?>(context) {
+        private val mContext: Context
+        private val mUri: Uri
+        private val mPromise: Promise
+        private val mOptions: ReadableMap
+
+        init {
+            mContext = context
+            mUri = uri
+            mPromise = promise
+            mOptions = options
+        }
+
+        override fun doInBackgroundGuarded(vararg params: Void?) {
+            val source = File(mUri.path)
+            var input: FileInputStream? = null
+            var output: OutputStream? = null
+            val mimeType = getMimeType(mUri.toString())
+            val isVideo = mimeType != null && mimeType.contains("video")
+            try {
+                val album = mOptions.getString("album")
+                val isAlbumPresent = !TextUtils.isEmpty(album)
+
+                // Android Q and above
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                    val mediaDetails = ContentValues()
+                    if (isAlbumPresent) {
+                        val relativePath = Environment.DIRECTORY_DCIM + File.separator + album
+                        mediaDetails.put(MediaStore.MediaColumns.RELATIVE_PATH, relativePath)
+                    }
+                    mediaDetails.put(MediaStore.MediaColumns.MIME_TYPE, mimeType)
+                    mediaDetails.put(MediaStore.Images.Media.DISPLAY_NAME, source.name)
+                    mediaDetails.put(MediaStore.Images.Media.IS_PENDING, 1)
+                    val resolver = mContext.contentResolver
+                    val mediaContentUri = if (isVideo) resolver.insert(
+                        MediaStore.Video.Media.EXTERNAL_CONTENT_URI,
+                        mediaDetails
+                    ) else resolver.insert(
+                        MediaStore.Images.Media.EXTERNAL_CONTENT_URI,
+                        mediaDetails
+                    )
+                    output = resolver.openOutputStream(mediaContentUri!!)
```

Comment:
`ContentResolver#insert()` might return nullable `mediaContentUri`, shall we check it instead of assertion`mediaContentUri!!`?

Reply:
I am not doing ANY manual code changes. The goal is to convert to kotlin with minimum changes and refactor/add null checks in follow-up PRs. That is, if this PR is reviewed, accepted and merged...

Reply:
Well noted. I think it would be better if we can get the PR merged firstly, and then, do the null-safety check for the whole project.